code/controllers/subsystem/traitor.dm

```
@@ -0,0 +1,66 @@
+	var/generate_objectives = TRUE
+
+/datum/controller/subsystem/traitor/Initialize(start_timeofday)
+	. = ..()
+	category_handler = new()
+	traitor_debug_panel = new(category_handler)
+
+/datum/controller/subsystem/traitor/fire(resumed)
+	var/player_count = length(GLOB.alive_player_list)
+	// Has a maximum of 1 minute, however the value can be lower if there are lower players than the ideal
+	// player count for a traitor to be threatening. Rounds to the nearest 10% of a minute to prevent weird
+	// values from appearing in the UI.
+	current_progression_scaling = max(min(
+		(player_count / CONFIG_GET(number/traitor_ideal_player_count)) * 1 MINUTES,
+		1 MINUTES
+	), 0.1 MINUTES)
+
+	var/progression_scaling_delta = (wait / (1 MINUTES)) * current_progression_scaling
+	var/previous_global_progression = current_global_progression
+
+	current_global_progression += progression_scaling_delta
+	for(var/datum/uplink_handler/handler in uplink_handlers)
+		if(!handler.has_progression || QDELETED(handler))
+			uplink_handlers -= handler
+		var/deviance = (previous_global_progression - handler.progression_points) / progression_scaling_deviance
+		if(abs(deviance) < 0.01)
+			// If deviance is less than 1%, just set them to the current global progression
+			// Prevents problems with precision errors.
+			handler.progression_points = current_global_progression
+		else
+			var/amount_to_give = progression_scaling_delta + (progression_scaling_delta * deviance)
+			amount_to_give = clamp(amount_to_give, 0, progression_scaling_delta * 2)
+			handler.progression_points += amount_to_give
+		handler.update_objectives()
+		handler.on_update()
+
+/datum/controller/subsystem/traitor/proc/register_uplink_handler(datum/uplink_handler/uplink_handler)
+	if(!uplink_handler.has_progression)
+		return
+	uplink_handlers |= uplink_handler
+	RegisterSignal(uplink_handler, COMSIG_PARENT_QDELETING, .proc/uplink_handler_deleted, TRUE)
```

1. `override = TRUE`, not just `TRUE`
2. All uses of `override = TRUE` should be documented as to ensure they're not just hiding an error.